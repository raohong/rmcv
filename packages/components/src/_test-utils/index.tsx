import { defaultTheme } from '@rmc-vant/system';

export * from '@testing-library/react';
export { default as sleep } from './sleep';

export const getTestThemes = () => defaultTheme;
