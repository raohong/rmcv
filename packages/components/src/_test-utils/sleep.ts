const sleep = (timeout = 0): Promise<void> =>
  new Promise<void>((r) => {
    setTimeout(() => {
      r();
    }, timeout);
  });

export default sleep;
