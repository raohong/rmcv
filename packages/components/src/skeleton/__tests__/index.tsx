import { render, screen } from '@testing-library/react';
import { Skeleton, skeletonClassNames } from '..';
import { getTestThemes } from '../../_test-utils';

const testId = 'skeleton';

it('render correctly', () => {
  const tree = render(<Skeleton>content</Skeleton>);

  expect(tree.asFragment()).toMatchSnapshot();
});

it('render with loading', () => {
  const com = render(<Skeleton data-testid={testId}>content</Skeleton>);

  expect(screen.getByTestId(testId)).not.toHaveTextContent('content');

  com.rerender(
    <Skeleton loading={false} data-testid={testId}>
      content
    </Skeleton>,
  );
  expect(screen.getByTestId(testId)).toHaveTextContent('content');
});

it('render with avatar', () => {
  render(<Skeleton data-testid={testId} avatar />);

  expect(
    screen.getByTestId(testId).querySelector(`.${skeletonClassNames.avatar}`),
  ).toBeInTheDocument();
});

it('render with avatarSize', () => {
  render(<Skeleton data-testid={testId} avatar avatarSize='2rem' />);

  expect(
    screen.getByTestId(testId).querySelector(`.${skeletonClassNames.avatar}`),
  ).toHaveStyleRule('width', '2rem');
});

it('render with round', () => {
  render(<Skeleton data-testid={testId} round />);

  expect(screen.getByTestId(testId)).toHaveClass(skeletonClassNames.round);
  expect(
    screen.getByTestId(testId).querySelector(`.${skeletonClassNames.title}`),
  ).toHaveStyleRule('border-radius', `${getTestThemes().radii.max}px`);
});

it('render with titleWidth', () => {
  render(<Skeleton data-testid={testId} titleWidth='50%' />);

  expect(
    screen.getByTestId(testId).querySelector(`.${skeletonClassNames.title}`),
  ).toHaveStyleRule('width', '50%');
});

it('render with row', () => {
  render(<Skeleton data-testid={testId} row={4} />);

  const rows = screen
    .getByTestId(testId)
    .querySelectorAll(`.${skeletonClassNames.paragraph}`);

  expect(rows.length).toBe(4);
});

it('render with rowWidth', () => {
  render(<Skeleton data-testid={testId} row={2} rowWidth={['10%', '20%']} />);

  const rows = screen
    .getByTestId(testId)
    .querySelectorAll(`.${skeletonClassNames.paragraph}`);

  expect(rows[0]).toHaveStyleRule('width', '10%');
  expect(rows[1]).toHaveStyleRule('width', '20%');
});
