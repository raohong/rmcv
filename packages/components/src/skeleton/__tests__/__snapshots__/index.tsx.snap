// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render correctly 1`] = `
<DocumentFragment>
  @keyframes animation-0 {
  50% {
    opacity: 0.5;
  }
}

.emotion-0 {
  box-sizing: border-box;
  margin: 0;
  padding: 0;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-tap-highlight-color: transparent;
  -webkit-font-smooth: antialiased;
  position: relative;
  touch-action: none;
  list-style: none;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  font-family: Open Sans,-apple-system,BlinkMacSystemFont,Helvetica Neue,Helvetica,Segoe UI,Arial,Roboto,PingFang SC,miui,Hiragino Sans GB,Microsoft Yahei,sans-serif;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 8px;
  -webkit-animation: animation-0 1.2s infinite;
  animation: animation-0 1.2s infinite;
}

.emotion-1 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  box-sizing: border-box;
  padding-top: 8px;
}

.emotion-2 {
  box-sizing: border-box;
  padding: 0;
  margin: 0;
  height: 16px;
  background: #f2f3f5;
  width: 40%;
}

.emotion-2:not(:first-of-type) {
  margin-top: 12px;
}

<div
    aria-live="assertive"
    class="RVuiSkeleton-root RVuiSkeleton-loading emotion-0"
    role="status"
  >
    <div
      class="emotion-1"
    >
      <h3
        class="RVuiSkeleton-title emotion-2"
      />
    </div>
  </div>
</DocumentFragment>
`;
